server.go

```diff
@@ -261,6 +271,160 @@ HANDLE:
 	goto LOOP
 }
 
+func (s *Server) handleUnlimitedPacket(frisbeeConn *Async, connCtx context.Context) {
+	p, err := frisbeeConn.ReadPacket()
+	if err != nil {
+		_ = frisbeeConn.Close()
+		s.onClosed(frisbeeConn, err)
+		return
+	}
+	if s.ConnContext != nil {
+		connCtx = s.ConnContext(connCtx, frisbeeConn)
+	}
+	wg := new(sync.WaitGroup)
+	closed := atomic.NewBool(false)
+	connCtx, cancel := context.WithCancel(connCtx)
+	handler := func(p *packet.Packet) {
+		handlerFunc := s.handlerTable[p.Metadata.Operation]
+		if handlerFunc != nil {
+			packetCtx := connCtx
+			if s.PacketContext != nil {
+				packetCtx = s.PacketContext(packetCtx, p)
+			}
+			outgoing, action := handlerFunc(packetCtx, p)
+			if outgoing != nil && outgoing.Metadata.ContentLength == uint32(len(*outgoing.Content)) {
+				s.preWrite()
+				err := frisbeeConn.WritePacket(outgoing)
+				if outgoing != p {
+					packet.Put(outgoing)
+				}
+				packet.Put(p)
+				if err != nil {
+					_ = frisbeeConn.Close()
+					if closed.CAS(false, true) {
+						s.onClosed(frisbeeConn, err)
+					}
+					cancel()
+					wg.Done()
+					return
+				}
+			} else {
+				packet.Put(p)
+			}
+			switch action {
+			case NONE:
+			case CLOSE:
+				_ = frisbeeConn.Close()
+				if closed.CAS(false, true) {
+					s.onClosed(frisbeeConn, nil)
+				}
+				cancel()
+			}
+		} else {
+			packet.Put(p)
+		}
+		wg.Done()
+		return
+	}
+HANDLE:
```

Comment:
I think `for {}` is more readable, easier to parse than GOTO

Reply:
Yep makes sense, the goto helped when we actually needed to jump around but the for loop should be just as fast.